src/vm/threads.h

```diff
@@ -4830,8 +4830,9 @@ class Thread: public IUnknown
     bool HasPendingGCStressInstructionUpdate()
     {
         LIMITED_METHOD_CONTRACT;
-        CONSISTENCY_CHECK((NULL == m_pbDestCode) == (NULL == m_pbSrcCode));
-        return m_pbDestCode != NULL;
+        BYTE* dest = VolatileLoad(&m_pbDestCode);
+        CONSISTENCY_CHECK((NULL == dest) || (NULL != VolatileLoadWithoutBarrier(&m_pbSrcCode)));
+        return dest != NULL;
```

Comment:
HasPendingGCStressInstructionUpdate() is called cross thread.  We must be careful about reading these pointers consistent with the order in which they were written.

+ Make only one volatile read of m_pbDestCode.  To determine whether there is a pending update.
+ Read m_pbSrcCode after m_pbDestCode to ensure it is set when needed.
+ Only check m_pbSrcCode when being set actually matters.

Assumes we are not racing with another thread clearing stress mode on this thread which would require more complicated logic.  This is currently a safe assumption.

Reply:
This revised consistency check failed once last night.  Looks like the assumption above must be false.  There must be a race to clear the stress mode.

I will implement the more complete solution.